fluxion/decoders/gstreamer.py

```diff
@@ -47,6 +50,20 @@ def decode(self, input_filepath: str, output_filepath: str):
                        stdout=subprocess.DEVNULL, check=True)
         return file_checksum(output_filepath)
 
+    @lru_cache
+    def check_run(self):
+        # pylint: disable=broad-except
+        elements = self.decoder_bin.split('!')
+        gst_inspect = f'gst-inspect-{self.gst_api}'
+        for element in elements:
+            element = element.strip().split()[0]
+            try:
+                subprocess.run([gst_inspect, element], stdout=subprocess.DEVNULL,
+                               stderr=subprocess.DEVNULL, check=True)
+            except Exception:
```

Comment:
If gst-launch-1.0 does not exists, an OSError is raised

Reply:
Yes, in which case the exception will be catched and we consider that the decoder cannot be run

Reply:
My point here is that you can catch OSError and enable back the broad-except pylint error

Reply:
Right, but we're not checking explicitly the exit code but relying that check=True will raise `subprocess.CalledProcessError`